src/java/org/apache/cassandra/utils/TimeUUID.java

```diff
@@ -293,6 +293,22 @@ public static String toString(TimeUUID ballot, String kind)
         return ballot == null ? "null" : String.format("%s(%d:%s)", kind, ballot.uuidTimestamp(), ballot);
     }
 
+    public int sequence()
+    {
+        return (int) ((lsb >> 48) & 0x0000000000003FFFL);
+    }
+
+    /**
+     * Returns a new TimeUUID with the same timestamp as this one, but with the provided sequence value.
```

Comment:
Let's call out that this method should be used with care as it removes "uniqueness" guarantees of the returned `TimeUUID` (same as we warn on `minAtUnixMillis` for instance).